src/Ecs/Systems/SkillMenuSystem.gd

```diff
@@ -0,0 +1,30 @@
+extends "res://Ecs/Core/System.gd"
+
+var _event_bus
+var _skillmenu 
+
+func _init(ui, event_bus):
+	_skillmenu  = ui.get_node("SkillMenu")
+	_event_bus = event_bus
+	_event_bus.connect("set_player", self, "on_set_player")
+
+func on_set_player(player):
+	if player.has("SkillComponent") and OS.has_touchscreen_ui_hint():
+		clear_skillmenu()
+		var skills = player.get("SkillComponent").skills
+		for skill in skills:
+			var index = skills.find(skill)
+			var skill_button = create_skill_button(skill, index)
+			skill_button.position.x = 64 * index
```

Comment:
This is a big assumption. A strategy I used in the past is to keep a map such as:

`const SKILL_INDEX_X = {
  "bow": 64,
  "energy shield": 128
}`

Then you can reference it as `skill_button.position.x = SKILL_INDEX_X[skill]` or something.

Reply:
> This is a big assumption

I'm sorry, what is? We're multiplying by the index, so they're ordered as they are in `SkillComponent`. Fixing them with a map isn't as generalised.